source/Calamari.Tests/Terraform/TerraformFixture.cs

```diff
@@ -0,0 +1,358 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Calamari.Commands.Support;
+using Calamari.Deployment;
+using Calamari.Integration.FileSystem;
+using Calamari.Terraform;
+using Calamari.Tests.Fixtures;
+using Calamari.Tests.Helpers;
+using FluentAssertions;
+using NUnit.Framework;
+using Octostache;
+using Newtonsoft.Json.Linq;
+
+namespace Calamari.Tests.Terraform
+{
+    [TestFixture]
+    [RequiresNonFreeBSDPlatform]
+    [RequiresNon32BitWindows]
+    [RequiresNonMac]
+    [RequiresNonMono]
+    [Category(TestCategory.CompatibleOS.Windows)]
+    public class TerraformFixture
+    {
+        private string customTerraformExecutable;
+
+        [OneTimeSetUp]
+        public async Task InstallTools()
+        {
+            var destinationDirectoryName = TestEnvironment.GetTestPath("TerraformCLIPath");
+
+            if (Directory.Exists(destinationDirectoryName))
+            {
+                var path = Directory.EnumerateFiles(destinationDirectoryName).FirstOrDefault();
+                if (path != null)
+                {
+                    customTerraformExecutable = path;
+                    Console.WriteLine($"Using existing terraform located in {customTerraformExecutable}");
+                    return;
+                }
+            }
+
+            Console.WriteLine("Downloading terraform cli...");
+
+            using (var client = new HttpClient())
```

Comment:
It would be good to put some retries around this to make it more resilient to network outages. It's been a source of flaky tests in other areas. A better option might be to commit the CLI version you want to test.

Reply:
The CLI is ~80Mbs, I didn't want to add such a big binary. But I can add the retries.